src/lib/optimizer/strategy/index_scan_rule.hpp

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "abstract_rule.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+class AbstractLQPNode;
+class PredicateNode;
+
+/**
+ * This optimizer rule finds PredicateNodes whose children are StoredTableNodes. These PredicateNodes are candidates
+ * for being executed by IndexScans. If the expected selectivity of the predicate falls below a certain threshold the
+ * ScanType of the PredicateNode is set to IndexScan.
```

Comment:
Add a comment that you rely on Predicate Reordering to get you the most promising PredicateNode to the bottom?

Reply:
Hm, it helps, but these rules are not necessarily connected. Connected together. Together connected.